financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -83,31 +79,29 @@ public IdempotencyValidationResult validateIdempotency(ConsentManageData consent
-                List<String> consentIds = IdempotencyValidationUtils
-                        .getConsentIdsFromIdempotencyKey(idempotencyKeyName, idempotencyKeyValue);
-                // Check whether the consent id list is not empty. If idempotency key exists in the database then
-                // the consent Id list will be not empty.
-                if (!consentIds.isEmpty()) {
-                    if (log.isDebugEnabled()) {
-                        log.debug(String.format("Idempotency Key  %s exists in the database. Hence this is an" +
-                                " idempotent request", idempotencyKeyValue.replaceAll("[\r\n]", "")));
-                    }
-                    for (String consentId : consentIds) {
-                        DetailedConsentResource consentResource = consentCoreService.getDetailedConsent(consentId);
-                        if (consentResource != null) {
-                            return validateIdempotencyConditions(consentManageData, consentResource);
-                        } else {
-                            String errorMsg = String.format(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS, consentId);
-                            log.error(errorMsg.replaceAll("[\r\n]", ""));
-                            throw new IdempotencyValidationException(errorMsg);
-                        }
+            // idempotencyKeyName is the name of the attribute key which the idempotency key
+            // is stored against in the consent attributes
+            String idempotencyKeyName = getIdempotencyAttributeName(consentManageData);
+            // Retrieve consent ids that have the idempotency key name and value as attribute
+            List<String> consentIds = IdempotencyValidationUtils
+                    .getConsentIdsFromIdempotencyKey(idempotencyKeyName, idempotencyKeyValue);
+            // Check whether the consent id list is not empty. If idempotency key exists in the database then
+            // the consent Id list will be not empty.
+            if (!consentIds.isEmpty()) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Idempotency Key  %s exists in the database. Hence this is an" +
+                            " idempotent request", idempotencyKeyValue.replaceAll("[\r\n]", "")));
+                }
+                for (String consentId : consentIds) {
+                    DetailedConsentResource consentResource = consentCoreService.getDetailedConsent(consentId);
+                    if (consentResource != null) {
+                        return validateIdempotencyConditions(consentManageData, consentResource);
+                    } else {
+                        String errorMsg = String.format(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS, consentId);
+                        log.error(errorMsg.replaceAll("[\r\n]", ""));
+                        throw new IdempotencyValidationException(errorMsg);
```

Comment:
Here what you have done is removing the payload check right? So now the idempotency validation will happen for both types of requests which contain a payload and do not contain a payload. Correct?

Reply:
yes for scenarios without payload also it has {} as the payload